src/BaseIO.cpp

```diff
@@ -31,22 +31,28 @@ const BaseDataType BaseDataType::DSTR = BaseDataType(T_STR, DEFAULT_STR_SIZE);
 
 // BaseIO
 
-BaseIO::BaseIO()
-    : readyToOpen(true)
-    , opened(false)
+BaseIO::BaseIO(const std::string& filename)
+    : m_filename(filename)
+    , m_readyToOpen(true)
+    , m_opened(false)
 {
 }
 
 BaseIO::~BaseIO() {}
 
 bool BaseIO::isOpen() const
 {
-  return opened;
+  return m_opened;
+}
+
+std::string BaseIO::getFileName()
+{
+  return this->m_filename;
```

Comment:
```suggestion
  return m_filename;
```

Reply:
Some getters are defined as inline functions in the `.hpp` file and some are defined in the `.cpp` file. Wondering if we want to choose one approach and make it consistent.

Reply:
For simple getter functions,I think it is generally useful to define them as inline methods in the header file as it: 1) reduces overhead and improves performance, because the compiler usually can expand inline functions directly at compile time so you don't have the overhead of the function call, and 2) it helps keep the code readable and avoid redundant text in the header and cpp file. For more complex getter functions that involve validation, performing data retrieval, or other complex operations, I think it is best to implement those in the cpp file, to keep the header file easy to read and also expanding complex functions at compile can become expansive and is not as effective.